csrc/scheduler/normalization_utils.cpp

```diff
@@ -601,6 +601,41 @@ bool isConnectedOnlyThroughReductionProducer(
   return true;
 }
 
+bool isReductionIterationAxisMatched(
+    const std::vector<TensorView*>& inner_reduction_tvs,
+    const std::vector<TensorView*>& outer_reduction_tvs) {
+  // set up reference, checkIfReductionsAreInnerOuter already ensures all the
+  // tensor domains are either iteration or reduction, so we can just use a
+  // vector of bool.
+  auto reference_tv = inner_reduction_tvs[0];
+  std::vector<bool> is_reduction(reference_tv->nDims(), false);
+  for (const auto i : c10::irange(reference_tv->nDims())) {
+    if (reference_tv->axis((int)i)->isReduction()) {
+      is_reduction[i] = true;
+    }
+  }
+  // check other inner reduction tvs, the corresponding axis should be
+  // reduction.
+  for (auto i : c10::irange(1, inner_reduction_tvs.size())) {
+    auto tv = inner_reduction_tvs[i];
+    for (const auto i : c10::irange(tv->nDims())) {
+      if (tv->axis((int)i)->isReduction() != is_reduction[i]) {
+        return false;
+      }
+    }
+  }
+  // check outer reduction tvs, the corresponding axis should be iteration.
+  for (auto tv : outer_reduction_tvs) {
+    for (const auto i : c10::irange(tv->nDims())) {
+      if (tv->axis((int)i)->isIteration() != is_reduction[i]) {
```

Comment:
What happens if one domain is a broadcast in both inner and outer tensors. Do we support tensors with broadcast domains? At this this condition should just pass. Not sure if that's a problem.

Reply:
`checkIfReductionsAreInnerOuter` already ensures all the tensor domains are either iteration or reduction. Will reject if there is a broadcast domain.

Reply:
OK, please assert that here.